velox/functions/sparksql/tests/ArrayInsertTest.cpp

```diff
@@ -0,0 +1,329 @@
+      {"[[1, 1], null, [3, 3]]", "[[5, null], [null, 6], [null, null], []]"});
+  auto expected = makeNestedArrayVectorFromJson<int64_t>(
+      {"[[0, 0], [1, 1], null, [3, 3]]",
+       "[[0, 0], [5, null], [null, 6], [null, null], []]"});
+  testExpression("array_insert(c0, 1, ARRAY[0, 0], false)", {arrays}, expected);
+
+  const auto input = makeRowVector(
+      {arrays, makeArrayVectorFromJson<int64_t>({"null", "null"})});
+  expected = makeNestedArrayVectorFromJson<int64_t>(
+      {"[null, [1, 1], null, [3, 3]]",
+       "[null, [5, null], [null, 6], [null, null], []]"});
+  testExpression("array_insert(c0, 1, c1, false)", input, expected);
+
+  expected = makeNestedArrayVectorFromJson<int64_t>(
+      {"[[1, 1], null, [0, 0], [3, 3]]",
+       "[[5, null], [null, 6], [null, null], [0, 0], []]"});
+  testExpression(
+      "array_insert(c0, -2, ARRAY[0, 0], false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, mapArrays) {
+  using M = std::vector<std::pair<int32_t, std::optional<int32_t>>>;
+  auto arrays = makeArrayOfMapVector<int32_t, int32_t>(
+      {{M{{1, 10}}}, {M{{2, 20}, {3, 30}}, std::nullopt}});
+  auto expected = makeArrayOfMapVector<int32_t, int32_t>(
+      {{M{{0, 0}}, M{{1, 10}}},
+       {M{{0, 0}}, M{{2, 20}, {3, 30}}, std::nullopt}});
+  testExpression("array_insert(c0, 1, map(0, 0), false)", {arrays}, expected);
+
+  expected = makeArrayOfMapVector<int32_t, int32_t>(
+      {{std::nullopt, M{{1, 10}}},
+       {std::nullopt, M{{2, 20}, {3, 30}}, std::nullopt}});
+  testExpression(
+      "array_insert(c0, 1, null::MAP(INTEGER, INTEGER), false)",
+      {arrays},
+      expected);
+
+  expected = makeArrayOfMapVector<int32_t, int32_t>(
+      {{M{{0, 0}}, M{{1, 10}}},
+       {M{{2, 20}, {3, 30}}, M{{0, 0}}, std::nullopt}});
+  testExpression("array_insert(c0, -2, map(0, 0), false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, structArrays) {
+  using ArraysOfRow =
+      std::vector<std::vector<std::optional<std::tuple<int32_t, int32_t>>>>;
+  auto rowType = ROW({{"a", INTEGER()}, {"b", INTEGER()}});
+  auto arrays = makeArrayOfRowVector(
+      ArraysOfRow{{{{1, 1}}}, {{{2, 2}}, {{3, 3}}}}, rowType);
+  auto expected = makeArrayOfRowVector(
+      ArraysOfRow{{{{0, 0}}, {{1, 1}}}, {{{0, 0}}, {{2, 2}}, {{3, 3}}}},
+      rowType);
+  testExpression(
+      "array_insert(c0, 1, row_constructor(0, 0), false)", {arrays}, expected);
+
+  expected = makeArrayOfRowVector(
+      ArraysOfRow{{std::nullopt, {{1, 1}}}, {std::nullopt, {{2, 2}}, {{3, 3}}}},
+      rowType);
+  testExpression(
+      "array_insert(c0, 1, null::STRUCT(a INTEGER, b INTEGER), false)",
+      {arrays},
+      expected);
+
+  expected = makeArrayOfRowVector(
+      ArraysOfRow{{{{0, 0}}, {{1, 1}}}, {{{2, 2}}, {{0, 0}}, {{3, 3}}}},
+      rowType);
+  testExpression(
+      "array_insert(c0, -2, row_constructor(0, 0), false)", {arrays}, expected);
+}
+
+TEST_F(ArrayInsertTest, nullInputs) {
+  // null input array
```

Comment:
nit: capitalize the first letter and add `.` at the end for comments. Ditto for the others.

Reply:
Thanks, updated.